packages/firebase/rollup.config.js

```diff
@@ -49,6 +49,20 @@ const plugins = [
 
 const external = Object.keys(pkg.dependencies || {});
 
+const uglifyOptions = {
+  mangle: {
+    properties: {
+      keep_quoted: true,
+      regex: /_$|^_/
+    }
+  },
+  compress: {
+    passes: 3,
+    unsafe: true,
```

Comment:
Does it mean it could break things in some case?

Reply:
The `compress.unsafe` option applies these 4 transformations:
- `new Array(1, 2, 3)` or `Array(1, 2, 3)` → `[ 1, 2, 3 ]`
- `new Object()` → `{}`
- `String(exp)` or `exp.toString()` → `"" + exp`
- `new Object/RegExp/Function/Error/Array (...)` → the `new` is discarded

These should actually be safe in *most* cases. That being said, I just did a build without that option and it seems like it only provides a very marginal reduction in size, about an extra 0.2% on average.

I don't think it's worth it, so I'll remove it.

Reply:
It looks great, thank you!